src/main/java/net/karlmartens/dotnet/Distribution.java

```diff
@@ -118,20 +120,42 @@ private void archive(File file) throws IOException {
         File targetFile = target.toFile();
         targetFile.createNewFile();
 
-        try (TarOutputStream out = new TarOutputStream(new GZIPOutputStream(new BufferedOutputStream(new FileOutputStream(targetFile, false))))) {
-            for (String includedFile : scanner.getIncludedFiles()) {
-                LOGGER.debug("Adding file {}", includedFile);
-                File f = source.resolve(includedFile).toFile();
+        switch(getExtension().getDistributeAs()) {

```

Comment:
This switch statement should be over an enumeration and the case implementations extracted into methods. 

```
OutputStream out = new FileOutputStream(targetFile, false);
switch(getExtension().getDistributeAs()) {
  case TAR:
    produceTar(scanner, out);
    break;

  case ZIP:
    produceZip(scanner, out);
    break;

  default:
    throw new IllegalStateException();
}
```
The above implementation would lead to a significant amount of duplicated code and based on the DRY principle a better implementation would be to introduce an abstraction for the archive and as we scan the included files add files to it. This might look like:

```
with (Archive archive = Archive.getInstance(getExtension().getDistributeAs(), targetFile)) {
    for (String includedFile : scanner.getIncludedFiles()) {
        LOGGER.debug("Adding file {}", includedFile);
        File f = source.resolve(includedFile).toFile();
        archive.add(f);
    }
}
```